tests/tests/compiler/futures/async_before_call_fail.leo

```diff
@@ -0,0 +1,22 @@
+/*
+namespace = "Compile"
+expectation = "Fail"
+*/
+
+program child.aleo {
+    transition t() {}
+}
+
+ // --- Next Program --- //
+
+import child.aleo;
+
+program parent.aleo {
+    async transition foo() -> Future {
+        let future: Future = finalize_foo();
+        child.aleo/t();
```

Comment:
I think it should be the case that (non-async) external transition calls can follow the async call. The more general rule is that all futures produced by external calls, must by consumed by a local async call. snarkVM has stricter restrictions that the async call must be last but the Leo compiler can re-order without changing semantics.